_transcripts/031-rust-1.46-1.47.md

```diff
@@ -0,0 +1,894 @@
+All you're committing to is that going forward, there will be implementations
+for arrays of arbitrary length. And the mechanism just happens to currently be
+const generics. But there's nothing in the, in what has been stabilized that
+requires that. So it might be that in the future, const generics just, like,
+imagine const generics does not work out, they could still provide some other
+built-in mechanism in the compiler that gives implementations for arrays of any
+length, and that would be sufficient to still fulfill the contract that's now
+essentially been stabilized.
+
+__Ben__: Stabilized in some sense, yeah. So again, actual stabilization of this
+feature is still forthcoming with no timeline. But there has been movement here
+in terms of the idea being, sort of trying to stabilize all of const generics
+was a pretty big thing. Trying to stabilize just a minimally useful subset of
+const generics, which now has its own feature flag at the compiler, the idea is
+to get all current uses compiling on this, to kind of like shake out any bugs,
+and I believe that's coming along pretty well. I look at the— there's a tracking
+issue for the `min_const_generics` feature flag, where they're going over,
+pretty recently, the remaining blocking issues and any bugs that are happening.
+So, yeah, follow that if you want const generics in your own code.
+
+__Jon__: Yeah, I mean— what I meant with stabilization here was more of, the
+stabilization of implementations for any length arrays. Like, that doesn't bind
+them to const generics, even though it seems pretty likely that const generics,
+or at least some part of it, is going to come hopefully, maybe soon.
+
+__Ben__: Yeah. Hopefully this year at least.
+
+__Jon__: One interesting point that we don't need to spend too much time on is
+the fact that most traits are now implemented for arrays of any length. But
+there's an exception, which is the `Default` trait. So they now have these
+generic implementations for any `N` for, like, `Debug` and `Clone` and `Copy`
+and whatnot. But `Default` is not among them. And the reason for this is that in
+the standard library today, like prior to 1.47, there's an— the implementation
+of `Default` for arrays of length zero do not require that the type of the array
+implements `Default`. So even if you have a type that doesn't implement
+`Default`, you can still get a zero-length array of that type, by using the
+`Default` trait.
+
+__Ben__: Which is totally reasonable, in a vacuum.
+
+__Jon__: Yeah, because you don't have any `T`s, so it doesn't matter if `T`
+implements `Default`. Unfortunately, there isn't, at least far as I'm aware, a
+way to express this in today's const generics, right? If you imagine that you
+write like `impl<T, const N> Default for [T; N]`. You don't have a way to say,
+like, require `Default` for `T` if `N` is greater than one, or greater than or
+equal to one, but don't require it if `N` is 0. You can't currently write that.
```

Comment:
Please review my translation of speech to Rust? (audio at 26:38)

What was actually spoken was "impl T and const generic N..." so I'm guessing this is what was meant.